### DISCUSSION TITLE
New document is not a text file?

### DISCUSSION THREAD

**Discussion by HubKing** [elementary/code#1411] [Q&A]
I installed it from Arch etc repository. Whenever I create a new tab, I see this thing below. If I click "Load anyway", it shows the document fine. Why do I see this?

![image](https://github.com/elementary/code/assets/10900489/20c7201f-28f7-46f7-9533-1bc51e0df613)

**Comment by jeremypw** [comment]
Is it natively installed or Flatpak? Could you give detailed steps to recreate the issue?  When you create a new tab (e.g. by pressing `+` , a temporary file should be created in `~/.local/share/io.elementary.code/unsaved` (if that exists in Arch) or somewhere else (not sure where) in Flatpak.  If you can locate this file what mimetype is it? I did not experience any problem creating New Documents running the latest version either natively or as a Flatpak in elementary 7.1.  You may get more information by running Code in a Terminal and looking at the message.  Running with the GTK_MESSAGES_DEBUG=all will give further info.

**Reply by HubKing** [reply]
* It's from the Arch "extra" repository, so "native".
* The file exists in that location: "unsaved".
* The mime type seems to be 
    ```
    file --mime-type 'Text file from 2024-01-22 06:15:42:297334.txt'
    Text file from 2024-01-22 06:15:42:297334.txt: inode/x-empty
    ```
* Double-clicking the text file in Gnome Files opens the text file with Gnome Text Editor.
* About that `DEBUG` thing, I don't know how I should do that. Where is the debug message?

**Reply by jeremypw** [reply]
The problem is the mime-type - Code assumes all text files (even empty ones) have a mime-type that is a type of "text/plain". On Ubuntu/elementary an empty file with an extension of ".txt" does have that mime-type (presumably on the basis of the extension).  However it seems that on Arch a different mime-type is returned hence the warning.  Code still successfully loads an empty file if "Load Anyway" is pressed.  An issue needs to be raised in the Code bug tracker (https://github.com/elementary/code/issues) - it would be easy to fix this by allowing the "inode/empty" mime-type to be loaded without warning.

**Reply by jeremypw** [reply]
Interestingly the `file --mime-type` command on elementary also gives `inode/x-empty` as the mime-type, However the Files properties window gives the mime-type as `text/plain`.  I think this is because of the way the GLib commands to determine content type and mime-type of a file works which must differ on the two OSes.  Both Files and Code determine the mimetype from the content-type returned from the file attribute `GLib.FileAttribute.STANDARD_CONTENT_TYPE`.
